server/app/views/admin/programs/ProgramBlockBaseView.java

```diff
@@ -0,0 +1,108 @@
+package views.admin.programs;
+
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.li;
+import static j2html.TagCreator.span;
+import static j2html.TagCreator.text;
+import static j2html.TagCreator.ul;
+
+import com.google.common.collect.ImmutableList;
+import j2html.tags.specialized.ButtonTag;
+import j2html.tags.specialized.DivTag;
+import j2html.tags.specialized.UlTag;
+import services.program.ProgramDefinition;
+import services.program.predicate.PredicateDefinition;
+import services.program.predicate.PredicateExpressionNode;
+import services.question.types.QuestionDefinition;
+import views.BaseHtmlView;
+import views.ViewUtils;
+import views.ViewUtils.ProgramDisplayType;
+import views.components.Icons;
+import views.style.AdminStyles;
+
+abstract class ProgramBlockBaseView extends BaseHtmlView {
```

Comment:
**nitpick (blocking):** Can you add a doc for this?  Something about providing utility methods to render the program UI boilerplate, etc

Reply:
Hi Shane, 

I agree that better documentation would be good. However as this pull request has been hanging for a long time and my second pull request depends on it too, I'd rather leave additional code cleanups of existing code for a separate change. I already created #4019 for cleanups in this area a while back to track some easy cleanups which are nice fillers for when I can't focus on more complex stuff.